### PR TITLE
add box selector

### LINKED ISSUE #3576: Box selector
**Is your feature request related to a problem? Please describe.**
A box selector for detection postprocessing is required

**Describe the solution you'd like**
Add box selector

**Describe alternatives you've considered**
N/A

**Additional context**
1. Top box selector
2. Non-maximum suppression selector

### LINKED ISSUE #4416: add batch_nms to box util
**Is your feature request related to a problem? Please describe.**
add batch_nms to box util

**Describe the solution you'd like**
- Please refer to batch_nms in
https://github.com/pytorch/vision/blob/main/torchvision/ops/boxes.py
- update NMS in detector into batch_nms

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Can-Zhao**
Signed-off-by: Can Zhao <canz@nvidia.com>

Fixes #3576. Fix #4416.

### Description
split box selection in detection inference out into BoxSelector
make box coder more general (not used in retinanet, but might be used in faster rcnn)
add batch nms

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — APPROVED** [review]
utilities for selecting candidate boxes

File: monai/apps/detection/utils/box_selector.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    """
+
+    def __init__(
+        self,
+        box_overlap_metric: Callable = box_iou,
+        apply_sigmoid: bool = True,
+        score_thresh=0.05,
+        topk_candidates_per_level=1000,
+        nms_thresh=0.5,
+        detections_per_img=300,
```
Comment (Nic-Ma) [inline monai/apps/detection/utils/box_selector.py:94]: Better to have typehint in these 4 args.

Thanks.

**Pull request merged**